### PR TITLE
better open_star

### PR DESCRIPTION
**Pull request by zlyfunction**
"Improve open_star so that it runs faster, and ensure that the resulting tuples point to the input simplex."

### CODE REVIEW

File: src/wmtk/SimplicialComplex.cpp [hunk 1]
```diff
@@ -287,15 +291,62 @@ SimplicialComplex SimplicialComplex::link(const Mesh& m, const Simplex& s)
...
+    constexpr PrimitiveType PV = PrimitiveType::Vertex;
+    constexpr PrimitiveType PE = PrimitiveType::Edge;
+    constexpr PrimitiveType PF = PrimitiveType::Face;
+    constexpr PrimitiveType PT = PrimitiveType::Tetrahedron;
+
+    auto sw = [&m](const Tuple& _t, const PrimitiveType& _ptype) {
+        return m.switch_tuple(_t, _ptype);
+    };
+
+    const long cell_dim = dynamic_cast<const TriMesh*>(&m) ? 2 : 3;
```
Comment (mtao) [inline src/wmtk/SimplicialComplex.cpp:305]: write this as a seperate functino for `open_star(const TriMesh& m, const Simplex& s)`

File: src/wmtk/SimplicialComplex.cpp [hunk 2]
```diff
@@ -287,15 +291,62 @@ SimplicialComplex SimplicialComplex::link(const Mesh& m, const Simplex& s)
...
+    const PrimitiveType s_ptype = s.primitive_type();
```
Comment (mtao) [inline src/wmtk/SimplicialComplex.cpp:306]: the previous dynamic cast to determine `cell_dim` and this `s_ptype` are redundant right?

**Review by mtao — APPROVED** [review]
What exactly did you mean to optimize here? Changes make sense at least

File: src/wmtk/MultiMeshManager.cpp [hunk 3]
```diff
@@ -140,6 +141,36 @@ namespace wmtk
         }
     }
 
+    std::vector<Simplex> MultiMeshManager::find_all_simplex_in_child_mesh(const Mesh& parent_mesh, long child_id, const Simplex& simplex_parent)
+    {
+        auto child_mesh_ptr = parent_mesh.multi_mesh_manager.child_meshes[child_id];
+        auto amp_to_child_handle = parent_mesh.multi_mesh_manager.map_to_child_handles[child_id];
+        PrimitiveType simplex_ptype = simplex_parent.primitive_type();
+        PrimitiveType childmesh_ptype = child_mesh_ptr->top_simplex_type();
```
[the remark this answers is not in the document]
Comment (zlyfunction) [reply src/wmtk/MultiMeshManager.cpp:149]: primitive_type is for a simplex, and top_simplex_type is for the top simplex in the mesh

**Pull request merged**